eslint.config.mjs

```diff
@@ -0,0 +1,218 @@
+// @ts-check
+import { FlatCompat } from '@eslint/eslintrc';
+import js from '@eslint/js';
+import eslintReact from '@eslint-react/eslint-plugin';
+import { globalIgnores } from 'eslint/config';
+import importX from 'eslint-plugin-import-x';
+import mocha from 'eslint-plugin-mocha';
+import noFloatingPromise from 'eslint-plugin-no-floating-promise';
+import noOnlyTests from 'eslint-plugin-no-only-tests';
+import youDontNeedLodashUnderscore from 'eslint-plugin-you-dont-need-lodash-underscore';
+import globals from 'globals';
+import tseslint from 'typescript-eslint';
+
+import prairielearn from '@prairielearn/eslint-plugin';
+
+const compat = new FlatCompat({ baseDirectory: import.meta.dirname });
+
+const NO_RESTRICTED_SYNTAX = [
+  {
+    selector:
+      'CallExpression[callee.type="MemberExpression"][callee.object.name="MathJax"][callee.property.name=/^(typeset|tex2chtml|tex2svg)$/]',
+    message: "Don't use the synchronous MathJax API; use a function like typesetPromise() instead.",
+  },
+  {
+    selector: 'MemberExpression[object.name="MathJax"][property.name="Hub"]',
+    message: 'Use MathJax.typesetPromise() instead of MathJax.Hub',
+  },
+  {
+    selector: 'ImportDeclaration[source.value="fs-extra"]:has(ImportNamespaceSpecifier)',
+    message: 'Use a default import instead of a namespace import for fs-extra',
+  },
+];
+
+export default tseslint.config([
+  js.configs.recommended,
+  tseslint.configs.stylistic,
+  tseslint.configs.strict,
+  eslintReact.configs['recommended-typescript'],
```

Comment:
I assume these are needed for preact, should this be restricted to assets? Not sure if this makes sense, please ignore if it doesn't.